wgpu-hal/src/dx12/device.rs

```diff
@@ -1092,7 +1179,7 @@ impl crate::Device<super::Api> for super::Device {
             },
             bind_group_infos,
             naga_options: hlsl::Options {
-                shader_model: hlsl::ShaderModel::V5_1,
```

Comment:
Does bumping this to 6.0 remove support for cards that support shadermodel 5.1?

According to techpowerup's gpu database that would mean removing support for Nvidia's Kepler (700 series, and some laptop 900 series) and earlier, as well as AMD's 1st gen GCN (old 7000 series, which had some cards rebranded and rereleased though the 200 and 300 series, and even some OEM only models in the 400 series and a single OEM only card in the 500 series).

Afaik all of those cards have had their driver support dropped at this point.

Reply:
Looked into this a bit, DXC only supports SM6.0 and higher, and FXC only supports up to SM5.1. DXC will upgrade anything it compiles below SM6.0 to SM6.0.

Do you know how other engines handle this, or what happens if you try running SM6.0 code on a SM5.1 gpu?
https://github.com/microsoft/DirectXShaderCompiler/issues/4038 seems like it's probably fine?

Reply:
If I'm understanding this correctly according to the (docs)[https://learn.microsoft.com/en-us/windows/win32/direct3d12/hardware-feature-levels], if a device supports `D3D_FEATURE_LEVEL_11_0` it should support shader model `6.0` and `5.1`, if this is incorrect, we can call CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL..) to check for shader support and choose FXC or DXC accordingly. 
unfortunately i don't have an old GPU to test this .

Reply:
It seems that only DXC CLI will upgrade SM5.1 to SM6.0, DXC lib would return an error, this won't be an issue if we take the solution i previously suggested.

Reply:
Looks like I was wrong about hardware support. GPU companies seemed to have added support for higher shadermodels in driver updates according to the DXC repo.

```
NVIDIA's r396 drivers (r397.64 and later) provide release mode support for DXIL 1.1 and Shader Model 6.1 on Win10 1709 and later, and experimental mode support for DXIL 1.2 and Shader Model 6.2 on Win10 1803 and later. These drivers also support DXR in experimental mode.

AMD’s driver (Radeon Software Adrenalin Edition 18.4.1 or later) provides release mode support for DXIL 1.1 and Shader Model 6.1.

Intel's 15.60 drivers (15.60.0.4849 and later) support release mode for DXIL 1.0 and Shader Model 6.0 as well as release mode for DXIL 1.1 and Shader Model 6.1 (View Instancing support only).
```

Reply:
We can assume that all DX12 devices support DXIL - DXBC is already being converted to DXIL by the d3d12 runtime behind the scenes.

Reply:
@cwfitzgerald `winapi` only exposes `SM6.0` and `SM5.1` , once wgpu starts using `windows-rs` we can start using the highest supported SM , supporting higher SM is a must so we can support other features like float16,Subgroup etc... ,
Using DXC will still be a big win even if we only support SM6.0 for now .

Reply:
The SM numbers are just enums, we can locally define the higher SMs